### PR TITLE
Fix parsing of Python tracebacks

### PR DESCRIPTION
**Pull request by andyhd**
## What

The code to nicely render Python tracebacks in dev was failing to parse `During handling of the above exception, another exception occurred` lines. This fixes that, which should make debugging easier.

### CODE REVIEW

File: app/api_clients/control_panel_api.js [hunk 1]
```diff
@@ -42,31 +42,39 @@ class DjangoError extends APIError {
   constructor(error) {
     super(error);
     this.original_message = this.message;
+    const pattern = /500 - "(.*)Request Method:/;
+    const match = this.original_message.match(pattern);
+    this.message = match[1]
+      .trim()
+      .replace(/\\n/g, '<br>\n');
```
Comment (r4vi) [inline app/api_clients/control_panel_api.js]: this is quite hairy parsing, can we have a test for it with example of input?

File: app/api_clients/control_panel_api.js [hunk 2]
```diff
@@ -42,31 +42,39 @@ class DjangoError extends APIError {
...
-      return { file, func, line_no, code };
-    });
-  }
+      match = line.match(pattern);
+      if (match) {
+        const [file, func, line_no, code] = match.slice(1);
+        return { file, func, line_no, code };
+      }
 
-  get message() {
```
Comment (xoen) [inline app/api_clients/control_panel_api.js]: This `message()` is not used anywhere right?

File: app/api_clients/control_panel_api.js [hunk 3]
```diff
@@ -42,31 +42,39 @@ class DjangoError extends APIError {
...
   }
 
   static match(error) {
     return error.message && error.message.indexOf('Traceback:') >= 0;
   }
 
   get python_traceback() {
-    let pattern = /\\nTraceback:\s+\\n\\n(.*)\\n\\nException Type/;
-    const match = this.original_message.match(pattern);
+    if (!this.original_message) {
```
Comment (xoen) [inline app/api_clients/control_panel_api.js]: Isn't `this.original_message` always there? I may miss something.

Comment (andyhd) [reply app/api_clients/control_panel_api.js]: It could be `null` or `undefined`

File: app/api_clients/control_panel_api.js (hunk 2 above)
Comment (andyhd) [reply app/api_clients/control_panel_api.js]: `get message()` has been replaced by `this.message` on line 47

File: app/api_clients/control_panel_api.js (hunk 1 above)
Comment (andyhd) [reply app/api_clients/control_panel_api.js]: Yeah, I'll write a test.

**Pull request merged**